workloads/spec-reference.md

```diff
@@ -0,0 +1,1095 @@
+    <td><code>annotations</code></td>
+    <td>
+     extra set of annotations to pass down to the APIDescriptor object.
+    </td>
+    <td>
+      <pre>
+      - name: annotations
+        value:
+          name: my-application
+          version: v1.2.3
+          team: store
+      </pre>
+    </td>
+  </tr>
+
+  <tr>
+    <td><code>api_descriptor</code></td>
+    <td>
+     information used to fill the state that you want of the APIDescriptor object
+    (its spec).
+    </td>
+    <td>
+      <pre>
+      - name: api_descriptor
+        value:
+          type: openapi
+          location:
+            baseURL: http://petclinic-hard-coded.my-apps.tapdemo.vmware.com/
+            path: "/v3/api"
+          owner: team-petclinic
+          system: pet-clinics
+          description: "example"
+      </pre>
+    </td>
+  </tr>
+</table>
+
+> **Note** it's required that the Workload include the
+> `apis.apps.tanzu.vmware.com/register-api: "true"` label to activate
+> this function.
+
+See [Use API Auto Registration](../api-auto-registration/usage.hbs.md) for more
+details about API auto registration.
+
+## config-writer (git or registry)
+
+The `config-writer` resource is responsible for performing the last mile of the
+supply chain: persisting in an external system (registry or git repository) the
+Kubernetes configuration generated throughout the supply chain.
+
+It can do so in three distinct manners:
+
+- publishing the configuration to a container image registry
+- publishing the configuration to a Git repository
+    - solely by using the push of a commit, or
+    - pushing a commit _and_ opening a pull request.
+
+Details about the different modes of operation are found in [Gitops vs
+RegistryOps](gitops-vs-regops.hbs.md) with the parameters documented in
+place.
+
+## deliverable
+
+The `deliverable` resource is responsible for creatiing a `Deliverable` object
+that represents the intention of delivering to the cluster the configuration
+that are produced by the supply chain.
+
+Parameters:
+
+<table>
+  <tr>
+    <th>parameter name</th>
+    <th>meaning</th>
+    <th>example</th>
+  </tr>
+
+  <tr>
+    <td><code>serviceAccount</code></td>
+    <td>
+      name of the serviceaccount (in the same namespace as the Deliverable) to
+      use for providing the necessary permissions to create the children
+      objects for deploying the objects created by the supply chain to the
+      cluster.
+    </td>
+    <td>
+      <pre>
+      - name: serviceAccount
+        value: default
+      </pre>
+    </td>
+  </tr>
+</table>
+
+> **Note** `--service-account` flag sets the `spec.serviceAccountName` key in
+> the Workload object. To configure the `serviceAccount` parameter, use
+> `--param serviceAccount=...`.
+
+> **Note** On build clusters where a corresponding `ClusterDelivery` doesn't
+> exist, the Deliverable takes no effect (similarly to a Workload without a
+> SupplyChain, no action is taken).
+
+## Deliverable Parameters reference
```

Comment:
Should this section be a sub section of 'deliverable' on line 827. As in should this be a H3 ###

Reply:
No, it stands on it's own. It just happens that the last items documented under the Workload parameters relates to deliverable.

I think during editing I may have just gotten the headings all wrong. This page should be titled with a H1 "Parameter Reference" at the top instead, and the current H1 "Workload Parameter Reference" should be demoted to a H2 so that it sits level with this "Deliverable Parameters reference" one.

Reply:
Ah ok, that makes sense, ok 
OK I've changed it so 
- there are just two H2 headings: Workload Parameter Reference and Deliverable Parameter reference
- updated the H1 as you indicated and added a new introduction underneath
-I restructured/reworded the table intro section a little, the table doesn't need it's own heading as is just part
of the introduction underneath the Workload Parameter Reference